src/lightning_app/structures/dict.py

```diff
@@ -64,10 +64,10 @@ def __setitem__(self, k, v):
         if isinstance(k, str) and "." in k:
             raise Exception(f"The provided name {k} contains . which is forbidden.")
 
+        _set_child_name(self, v, k)
```

Comment:
Can u share the failure trace + add a test for this example? If I understand properly, the issue was the works weren't named properly right?

Reply:
I can add a test, but this is broken on the latest release, so it would be best to get the fix in the release then follow-up with a test.